Add removeData parameter to ExApp Remove UI

edward-ly opened the pull request:
Closes #297.

edward-ly commented:
I noticed while making the PR that if there is more than one ExApp installed and disabled, the value of the checkbox persists when switching between apps in the list, and resets after closing and reopening the sidebar or leaving the page. I just want to confirm if this behavior is intended (or, if not, what the correct behavior should be).

bigcat88 commented on src/store/apps.js:
```diff
@@ -295,10 +295,10 @@ const actions = {
 		return api.requireAdmin().then((response) => {
 			context.commit('startLoading', appId)
-			return api.get(generateUrl(`/apps/app_api/apps/uninstall/${appId}`))
+			return api.get(generateUrl(`/apps/app_api/apps/uninstall/${appId}/${removeData}`))
```
if there is no strong reason to make it a path parameter, please, change it to query parameter

edward-ly replied:
As in the URL string should be `/apps/app_api/apps/uninstall/${appId}?removeData=${removeData}` instead?
That's the only other way I can think of; I don't quite understand how the `generateUrl` function is able to add query parameters otherwise (I thought it was the [`params`](https://nextcloud-libraries.github.io/nextcloud-router/functions/generateUrl.html) argument at first, but that turned out to be wrong).

bigcat88 commented:
> I noticed while making the PR that if there is more than one ExApp installed and disabled, the value of the checkbox persists when switching between apps in the list, and resets after closing and reopening the sidebar or leaving the page. I just want to confirm if this behavior is intended (or, if not, what the correct behavior should be).

it should always be in an unchecked state when switching between apps in the list (even if it was checked, then the focus was switched to another ExApp and switched back)

edward-ly commented:
> it should always be in an unchecked state when switching between apps in the list (even if it was checked, then the focus was switched to another ExApp and switched back)

Now added in fdf70df.

bigcat88 commented:
Thanks for the pull request @edward-ly
